src/librustc/hir/lowering.rs

```diff
@@ -1865,20 +1914,32 @@ impl<'a> LoweringContext<'a> {
         )
     }
 
-    fn lower_local(&mut self, l: &Local) -> P<hir::Local> {
+    fn lower_local(&mut self, l: &Local) -> (P<hir::Local>, OneVector<hir::ItemId>) {
         let LoweredNodeId { node_id, hir_id } = self.lower_node_id(l.id);
-        P(hir::Local {
+        let mut ids = OneVector::<hir::ItemId>::new();
+        if let Some(ref ty) = l.ty {
+            let mut visitor = ImplTraitTypeIdVisitor { ids: &mut ids };
+            visitor.visit_ty(ty);
+        }
+        let impl_trait_owner_id = *self.current_impl_trait_owner.last().unwrap();
```

Comment:
may `self.current_impl_trait_owner.last().map(|id| *id)` be slightly better?

Reply:
If there were a need for the actual `Option`, then yes... as it is, I don't think so?